src/blockchain/ain/whale/whale-client.ts

```diff
@@ -16,6 +16,10 @@ export class WhaleClient {
     return new WhaleApiClient(GetConfig().whale);
   }
 
+  async getAllUnspent(address: string): Promise<AddressUnspent[]> {
+    return await this.client.address.listTransactionUnspent(address);
+  }
+
   async getUnspent(address: string, expectedAmount: BigNumber): Promise<AddressUnspent[]> {
```

Comment:
Why is the return value an array?

Reply:
We could remove that here, but we would anyway need to make an array again in `jellyfish.service` as we want to work with the possibility of multiple inputs. It is not super clean, can change it, if needed :)

Reply:
Just looked again through the code, we have on line 120 and 121 either use `getAllUnspent` or `getUnspent` it would be more convenient if both return an array. I will keep it like it currently is